org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -387,8 +458,93 @@ public boolean accept(File dir, String name) {
 		return shouldSynchronize;
 	}
 
+	public static boolean upgradeGradleVersion(String projectDir, IProgressMonitor monitor) {
+		StringBuilder newUrlBuilder = new StringBuilder();
+		newUrlBuilder.append("https://services.gradle.org/distributions/gradle-");
+		newUrlBuilder.append(GradleCompatibilityChecker.CURRENT_GRADLE);
+		newUrlBuilder.append("-bin.zip");
+		String newDistributionUrl = newUrlBuilder.toString();
+		Path projectFolder = Paths.get(projectDir);
+		File propertiesFile = projectFolder.resolve("gradle").resolve("wrapper").resolve("gradle-wrapper.properties").toFile();
+		Properties properties = new Properties();
+		if (propertiesFile.exists()) {
+			try (FileInputStream stream = new FileInputStream(propertiesFile)) {
+				properties.load(stream);
+				properties.setProperty("distributionUrl", newDistributionUrl);
+			} catch (IOException e) {
+				return false;
+			}
+		} else {
+			properties.setProperty("distributionBase", "GRADLE_USER_HOME");
+			properties.setProperty("distributionPath", "wrapper/dists");
+			properties.setProperty("distributionUrl", newDistributionUrl);
+			properties.setProperty("zipStoreBase", "GRADLE_USER_HOME");
+			properties.setProperty("zipStorePath", "wrapper/dists");
+		}
+		try {
+			properties.store(new FileOutputStream(propertiesFile), null);
+		} catch (Exception e) {
+			return false;
+		}
+		BuildConfiguration build = getBuildConfiguration(projectFolder);
+		GradleBuild gradleBuild = GradleCore.getWorkspace().createBuild(build);
+		try {
+			gradleBuild.withConnection(connection -> {
+				connection.newBuild().forTasks("wrapper").run();
```

Comment:
```shell
gradle wrapper --gradle-version 7.3.1 --distribution-type all
```
You can pass the wrapper version argument in forTasks directly, no need to create "gradle-wrapper.properties" file in advance.

Reply:
If we keep the properties file unchanged and try to execute the `wrapper` task, the Gradle will try to configure the project with the old Gradle version before executing this task, and it will result in the same compatibility error. So we should modify the properties file manually in advance.